WORK IN PROGRESS - Solidity standalone

bobsummerwill opened the pull request:
DO NOT COMMIT.

This is work-in-progress on making the solidity repo completely standalone.    All the main executables build, but further cleanup is required.   Also need to work on the tests.

The approach was to orphan copies of the entire `webthree-helpers/cmake` and `libweb3core/libdevcore` folders into this repo, fix up the relative paths to get it working, and them to trim away at it from there and see how minimal we could get it.

bobsummerwill committed:
Drop full copies of cmake/ and libdevcore/ into solidity repo.
I'll get that working, so we have a standalone, repository, and will then trim it back from there.

bobsummerwill committed:
Add gitignore entry for CLion IDE temporary files (.idea)

bobsummerwill committed:
Ugly first pass, which builds.

bobsummerwill committed:
Deleted a bunch of files from our orphaned copy of libdevcore which we don't use.
These were the easy cuts.

bobsummerwill committed:
Minimize CommonIO.

bobsummerwill committed:
Stripped out all the excess cmake helper files.

bobsummerwill committed:
Chop out more stuff.   Looks like we're not doing any database stuff in solidity.

bobsummerwill committed:
Fixed relative path to self.

bobsummerwill committed:
More cleanup, of WindowsSDK and self-dependencies.

bobsummerwill committed:
Fixed whitespace in gitignore, which was missing line ending at end of file.

bobsummerwill committed:
Comment about the problem we have with extdep pattern when we have standalone repos.
Time for chocolatey right now?

bobsummerwill committed:
Re-added pthread dependency for Linux builds of solidity.
It was previously "not MSVC", but it looks like we only need it for Linux, not for OS X.
The same fix might even be possible upstream.   I should try that.

VoR0220 commented:
👍

chriseth commented:
This is fine for testing, but I think we should go the other way round:
Solidity does not make use of much inside libdevcore, it mainly uses sha3, FixedHash, bytesref and some convenience formatting and vector functions. We should copy sha3, create a slimmed-down version of fixedhash and either get rid of bytesref or try to also copy over a slimmed-down version of it.

I think if we start copying large amounts of code, maintainability will actually rather decrease than increase (even if we "promise" to only copy temporarily).

bobsummerwill commented:
Yeah - I'm just exploring my way to an answer on what our actual dependencies are, by chopping the CMake and libdevcore files down to the bare minimum, and then seeing what we have left.

See, for example ...

https://github.com/ethereum/solidity/pull/525/commits/93984fe1d209c37679f35dedb00a955e3a07d471
https://github.com/ethereum/solidity/pull/525/commits/202c3f6ed704b2777cf3f00fff8b06696740e704

I have plenty more chopping to do yet here!

bobsummerwill committed:
Chopped out most of the code from RLP.

Our only remaining API is ...

 inline bytes rlpList() { return RLPStream(0).out(); }

And the rest of internal code, which can be much simplified, because it's just going to be the result of that function which will be a constant.

bobsummerwill committed:
Shove all the RLP code into the source file.

bobsummerwill committed:
Chopped out almost all of the logging implementation (but not the API).

bobsummerwill committed:
Chopped out unused base36 code.

bobsummerwill committed:
Delete CommonJS, which appears not to be used.

bobsummerwill committed:
Only a little bit of CommonIO has strippable.
Lots of inlines.

bobsummerwill committed:
Chopped out a lot of Common.cpp/Common.h.
Most of what is left is pretty bad, IMHO.
Having global typedefs is not a good pattern in the first place, and the tensity of the names just makes these a maintenance issue.
Would probably be better to push these types up into the code which uses them?

bobsummerwill committed:
Chop out logging.   OK as per Christian.

bobsummerwill committed:
Worked through CommonData, most of which appears to be used.

bobsummerwill committed:
First pass at replicating the TravisCI, CircleCI and Appveyor automation I did for cpp-ethereum for solidity.

bobsummerwill committed:
Removed stray trailing backslash.

bobsummerwill committed:
Added "extdep" to standalone solidity.
We cannot actually have these "fake Windows packages" in both places, but this can be a workaround for now.

bobsummerwill committed:
OK - let's start with the full setup.sh from cpp-ethereum, and then chop it down from there.

bobsummerwill committed:
Try to minimize solidity setup.sh dependencies.

bobsummerwill committed:
Forgot to remove libjson-rpc-cpp from the list of dependencies on OS X.

bobsummerwill committed:
Removed curl from Linux setup (only used for RPC).

bobsummerwill committed:
Hmm.  Maybe we don't need CryptoPP or GMP either?

bobsummerwill committed:
Refreshed the CircleCI, TravisCI and Appveyor yml configuration files for standalone Solidity.
Bringing them up-to-date with all the progress made in cpp-ethereum, including switching to VS2015.
Added "deps" sub-module.

bobsummerwill committed:
Flush relevant webthree-helpers changes across into the orphaned "cmake" folder inside solidity standalone.
This should fix the paths such that VS2015 starts working.   I hope!

bobsummerwill committed:
Copied the full setup.sh from cpp-ethereum.
Want to get CircleCI builds working here, and then I will strip it back down again.
Might just be a caching issue?

bobsummerwill committed:
Stripped back out some WindowsSDK and Qt bits in the cmake files which we definitely don't need here for Solidity.

bobsummerwill committed:
Touch Solidity, to trigger CircleCI build.
The setup.sh still needs stripping right back.

bobsummerwill committed:
Stripped setup.sh back to just what it should be.
It looks like the remaining CircleCI issues are just some nasty caching thing.
Which may or may not be fixable.

bobsummerwill committed:
Deleted CircleCI support.
It is proving too unreliable to be worthwhile to maintain.
Caching issues.   No "matrix" ability.

bobsummerwill committed:
Strip LLVM PPA, which we don't need for Solidity.
It has actually been taken down temporarily too, but that just highlighted that we shouldn't have it here anyway.

bobsummerwill committed:
Merge branch 'develop' into solidity_standalone

# Conflicts:
#	.travis.yml

bobsummerwill committed:
Delete extdep folder, which is redundant now.
It was only needed for VS2013.

bobsummerwill committed:
Big messy splat of extra dependencies dragged in by the solidity tests.
The tests still don't quite build, but are probably close.
Disabled TESTS in the automation so this commit can go in and be tested for solc build.
Next step is to get the tests building and running cleanly, and then to prune back the dependencies to something rational.

bobsummerwill committed:
Debug text for noexcept() issue

bobsummerwill committed:
Removed debug text.   Looks like the Solidity break was just a clean build issue of some type.
